src/main/java/gr8pefish/ironbackpacks/util/registry/BackpackRegistry.java

```diff
@@ -0,0 +1,220 @@
+         * @return builder instance for chaining
+         * @since 4.0.0
+         */
+        public Builder colorProvider(StackColorProvider colorProvider) {
+            this.colorProvider = colorProvider;
+            return this;
+        }
+
+        /**
+         * Sets the default color provider for this backpack. Only applicable
+         * to dyeable backpacks.
+         * <p>
+         * Provides the default color for a backpack when undyed. Defaults to
+         * vanilla Minecraft behavior for dyeable items, i.e. leather armor -
+         * the color is leather armor brown (<tt>#A06540</tt>).
+         *
+         * @param defaultColorProvider default color provider implementation
+         * @return builder instance for chaining
+         * @since 4.0.0
+         */
+        public Builder defaultColorProvider(StackColorProvider defaultColorProvider) {
+            this.defaultColorProvider = defaultColorProvider;
+            return this;
+        }
+
+        /**
+         * Sets the item settings to pass to the item on creation. Defaults to
+         * being categorized under Tools in the creative menu. The created item
+         * will always be non-stackable.
+         *
+         * @param settings item settings
+         * @return builder instance for chaining
+         * @since 4.0.0
+         */
+        public Builder settings(Item.Settings settings) {
+            this.settings = settings;
+            return this;
+        }
+
+        private BaseBackpackItem build() {
+            if (dyeable) {
+                return new BaseDyeableBackpackItem(settings.maxCount(1), tier, upgradePoints, cols, rows,
+                        colorProvider, defaultColorProvider);
+            } else {
+                return new BaseBackpackItem(settings.maxCount(1), tier, upgradePoints, cols, rows);
+            }
+        }
+
+        /**
+         * Registers the item specification as an actual item. This is the terminal
+         * operation in a builder chain.
+         *
+         * @param id identifier to register the item under
+         * @since 4.0.0
+         */
+        public void register(Identifier id) {
+            BackpackRegistry.register(id, build());
+        }
+    }
+
+    /**
+     * Instantiates a builder for a backpack.
+     *
+     * @see Builder
+     * @return builder instance
+     * @since 4.0.0
+     */
+    public static Builder create() {
+        return new Builder();
+    }
+
+    /**
+     * Registers a backpack item directly. You should use this if your
+     * backpack item implementation differs from base.
+     *
+     * @param id identifier for the item
+     * @param item backpack item
+     * @since 4.0.0
+     */
+    public static void register(Identifier id, BackpackItem item) {
+        if (BACKPACKS.putIfAbsent(id, item) != null) {
+            LOGGER.warn("Backpack with ID " + id.toString() + " already exists! Skipping...");
```

Comment:
Logger has it's own form of string formatting which should be used.
```suggestion
            LOGGER.warn("Backpack with ID {} already exists! Skipping...", id);
```